tests/test_credentials.py

```diff
@@ -0,0 +1,235 @@
+import random
+import string
+
+import pytest
+
+import redis
+from redis import AuthenticationError, DataError, ResponseError
+from redis.credentials import CredentialProvider, StaticCredentialProvider
+from redis.utils import str_if_bytes
+from tests.conftest import _get_client, skip_if_redis_enterprise
+
+
+def init_acl_user(r, request, username, password):
+    # reset the user
+    r.acl_deluser(username)
+    if password:
+        assert (
+            r.acl_setuser(
+                username,
+                enabled=True,
+                passwords=["+" + password],
+                keys="~*",
+                commands=[
+                    "+ping",
+                    "+command",
+                    "+info",
+                    "+select",
+                    "+flushdb",
+                    "+cluster",
+                ],
+            )
+            is True
+        )
+    else:
+        assert (
+            r.acl_setuser(
+                username,
+                enabled=True,
+                keys="~*",
+                commands=[
+                    "+ping",
+                    "+command",
+                    "+info",
+                    "+select",
+                    "+flushdb",
+                    "+cluster",
+                ],
+                nopass=True,
+            )
+            is True
+        )
+
+    if request is not None:
+
+        def teardown():
+            r.acl_deluser(username)
+
+        request.addfinalizer(teardown)
+
+
+def init_required_pass(r, request, password):
+    r.config_set("requirepass", password)
+
+    def teardown():
+        try:
+            r.auth(password)
+        except (ResponseError, AuthenticationError):
+            r.auth("default", "")
+        r.config_set("requirepass", "")
+
+    request.addfinalizer(teardown)
+
+
+class TestCredentialsProvider:
+    @skip_if_redis_enterprise()
+    def test_credential_provider_without_supplier_only_pass(self, r, request):
+        # test for default user (`username` is supposed to be optional)
+        password = "password"
+        creds_provider = CredentialProvider(password=password)
+        init_required_pass(r, request, password)
+        assert r.auth(creds_provider.password) is True
+
+        r2 = _get_client(
+            redis.Redis, request, flushdb=False, credential_provider=creds_provider
+        )
+
+        assert r2.ping()
```

Comment:
Should this not be 
```suggestion
        assert r2.ping() is True
```
to match the rest of the tests?